File: imageeditlibrary/src/main/java/com/xinlan/imageeditlibrary/editimage/adapter/StickerAdapter.java
package com.xinlan.imageeditlibrary.editimage.adapter;

import java.io.File;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import androidx.recyclerview.widget.RecyclerView.ViewHolder;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.View.OnClickListener;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.nostra13.universalimageloader.core.DisplayImageOptions;
import com.nostra13.universalimageloader.core.ImageLoader;
import com.xinlan.imageeditlibrary.R;
import com.xinlan.imageeditlibrary.editimage.fragment.StickerFragment;


/**
 * 贴图分类列表Adapter
 *
 * @author panyi
 */
public class StickerAdapter extends RecyclerView.Adapter<StickerAdapter.ViewHolder> {
    private final Context mContext;
    private final int [] mBitmapInfo;
    public StickerAdapter(Context context,int[] bitmapInfo,OnImageViewClicked onImageViewClicked){
        this.mContext = context;
        this.mBitmapInfo = bitmapInfo;
        this.onImageViewClicked = onImageViewClicked;
    }
    public interface OnImageViewClicked{
        void imageViewClicked(int id);
    }
    private final OnImageViewClicked onImageViewClicked;

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        final  View view = LayoutInflater.from(mContext).inflate(R.layout.view_sticker_item,parent,false) ;
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.imageView.setImageResource(mBitmapInfo[holder.getAdapterPosition()]);
        holder.imageView.setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View v) {
                onImageViewClicked.imageViewClicked(mBitmapInfo[holder.getAdapterPosition()]);
            }
        });
    }

    @Override
    public int getItemCount() {
        return mBitmapInfo.length;
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        ImageView imageView ;
        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            imageView = itemView.findViewById(R.id.img);
        }
    }
}

File: app/src/main/java/com/weilai/jigsawpuzzle/adapter/special/StickerAdapter.kt
package com.weilai.jigsawpuzzle.adapter.special

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.weilai.jigsawpuzzle.R
import com.weilai.jigsawpuzzle.adapter.RecyclerViewHolder
import com.weilai.jigsawpuzzle.bean.StickerBean

class StickerAdapter : RecyclerView.Adapter<RecyclerViewHolder>() {
    private var mList : Array<StickerBean> = arrayOf()
    private var callback : StickerCallback? = null
    private lateinit var mContext: Context
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerViewHolder {
        mContext = parent.context
        return RecyclerViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.item_sticker,parent,false))
    }

    override fun onBindViewHolder(holder: RecyclerViewHolder, position: Int) {
        holder.setImage(R.id.img,mList[position].showResId)
            holder.getView<ImageView>(R.id.vip_tip).visibility = View.GONE
        holder.itemView.setOnClickListener {
            callback?.clickStickerItem(mList[position])
        }
    }

    override fun getItemCount(): Int {
        return mList.size
    }

    fun setData(list : Array<StickerBean>){
        this.mList = list
        notifyDataSetChanged()
    }

    fun setCallback(callback : StickerCallback){
        this.callback = callback
    }

    interface StickerCallback{
        fun clickStickerItem(stickerBean: StickerBean)
    }
}